src/Controller.ino

```diff
@@ -245,6 +245,10 @@ bool MQTTCheck(int controller_idx)
     {
       if (MQTTclient_should_reconnect) {
         addLog(LOG_LEVEL_ERROR, F("MQTT : Intentional reconnect"));
+        // Homie 3 & 4dev Controller Plugin
+        // Connect first and call all installed controller to publish autodiscover data
+        // hope this is the right place ;)
+        if (MQTTConnect(controller_idx)) CPluginCall(CPLUGIN_GOT_CONNECTED, 0);
```

Comment:
You're still changing the existing connect routines for the MQTT controllers.
Outside this if scope is also a call to MQTTConnect. (`return MQTTConnect` ) Why not change that part to not immediately return the connect state but call the `CPluginCall(CPLUGIN_GOT_CONNECTED)` from inside that `MQTTConnect` function.
Then you know for sure it will always be called if a connect has been made.

Reply:
I think I misunderstood you, sorry. Moved it to here. Think it is a "safe" place right after LWT is published.

```
  if (MQTTclient.publish(LWTTopic.c_str(), LWTMessageConnect.c_str(), 1)) {
    updateMQTTclient_connected();
    statusLED(true);
    mqtt_reconnect_count = 0;
    // call all installed controller to publish autodiscover data
    CPluginCall(CPLUGIN_GOT_CONNECTED, 0);
    return true; // end loop if succesfull
  }
```
 
Should I commit the change or wait for further comments?

Reply:
I forgot your comment before, now I understood:

```
  if (MQTTclient.publish(LWTTopic.c_str(), LWTMessageConnect.c_str(), 1)) {
    updateMQTTclient_connected();
    statusLED(true);
    mqtt_reconnect_count = 0;
    // call all installed controller to publish autodiscover data
    if (MQTTclient_should_reconnect) CPluginCall(CPLUGIN_GOT_CONNECTED, 0);
    return true; // end loop if succesfull
  }
```

Reply:
> I think I misunderstood you, sorry. Moved it to here. Think it is a "safe" place right after LWT is published.
> 
> ```
>   if (MQTTclient.publish(LWTTopic.c_str(), LWTMessageConnect.c_str(), 1)) {
>     updateMQTTclient_connected();
>     statusLED(true);
>     mqtt_reconnect_count = 0;
>     // call all installed controller to publish autodiscover data
>     CPluginCall(CPLUGIN_GOT_CONNECTED, 0);
>     return true; // end loop if succesfull
>   }
> ```
> Should I commit the change or wait for further comments?

This one is what I meant.
You don't have to check anymore whether it was a requested reconnect or one due to connection lost. It is a reconnect, so you should call the `CPLUGIN_GOT_CONNECTED`.
And yes please add a new commit to this branch where this pull request is referring to, then it will be added to the pull request.